### PR TITLE
fix: Fix max timestamp value comparison in chunk metadata

### PR DESCRIPTION
**Pull request by alamb**
Fixes https://github.com/influxdata/influxdb_iox/issues/3450

# Background
A `TimestampRange` structure was being used to represent *both* a predicate (where the bound is exclusive) and also set of a min/max values of data in mutable buffer and parquet chunks, where the bound is *inclusive*)

So in this specific case, we have a data min and max values of

```
946684801000000000 - 946684803000000000
```

But a predicate range of

```
946684801000000000 - 946684803000000001
```

The desired test is "are there any values in the data range that fall within the predicate range?". Clearly the answer to this question should be "yes"

However, the way the code is written it is computing "not is range disjoint with other" which was returning false.

# Fix
Create a separate `TimestampMinMax` struct to try and make it clear the semantics are (very) subtlety different.

# InfluxQL test update:
The `TestServer_WhereTimeInclusive` suite now passes (previously `TestServer_WhereTimeInclusive/last_GTE/LTE` was failing)
 
```
--- PASS: TestServer_WhereTimeInclusive (1.42s)
    --- PASS: TestServer_WhereTimeInclusive/all_GTE/LTE (0.02s)
    --- PASS: TestServer_WhereTimeInclusive/all_GTE (0.02s)
    --- PASS: TestServer_WhereTimeInclusive/all_LTE (0.02s)
    --- PASS: TestServer_WhereTimeInclusive/first_GTE/LTE (0.02s)
    --- PASS: TestServer_WhereTimeInclusive/last_GTE/LTE (0.02s)
    --- PASS: TestServer_WhereTimeInclusive/before_GTE/LTE (0.01s)
    --- PASS: TestServer_WhereTimeInclusive/all_GT/LT (0.01s)
    --- PASS: TestServer_WhereTimeInclusive/first_GT/LT (0.02s)
    --- PASS: TestServer_WhereTimeInclusive/last_GT/LT (0.02s)
    --- PASS: TestServer_WhereTimeInclusive/all_GT (0.02s)
    --- PASS: TestServer_WhereTimeInclusive/all_LT (0.02s)
```

There are now 11 failing tests (down from 14) 🎉 
```
(arrow_dev) alamb@MacBook-Pro-2:~/Software/idpe$ cat /tmp/influxql.log | grep -v AAL | grep -v 'logger\.go' | grep '^--- FAIL'  | wc -l
      11
```

### CODE REVIEW

File: mutable_buffer/src/snapshot.rs [hunk 1]
```diff
@@ -144,10 +145,10 @@ impl ChunkSnapshot {
         self.schema
             .find_index_of(TIME_COLUMN_NAME)
             .and_then(|idx| match &self.summary.columns[idx].stats {
-                Statistics::I64(stats) => Some(
-                    !TimestampRange::new(stats.min? as _, stats.max? as _)
-                        .disjoint(timestamp_range),
```
Comment (alamb) [inline mutable_buffer/src/snapshot.rs:149]: this `!disjoint` check was returning the wrong value when `stat.max == timestamp_range.max+1`

File: parquet_file/src/chunk.rs [hunk 2]
```diff
@@ -83,9 +83,9 @@ pub struct ParquetChunk {
     /// Schema that goes with this table's parquet file
     schema: Arc<Schema>,
 
-    /// Timestamp range of this table's parquet file
-    /// (extracted from TableSummary)
-    timestamp_range: Option<TimestampRange>,
+    /// min/max time range of this table's parquet file
+    /// (extracted from TableSummary), if known
+    timestamp_min_max: Option<TimestampMinMax>,
```
Comment (alamb) [inline parquet_file/src/chunk.rs:88]: Changed to use `TimestampMinMax`

**Comment by alamb** [comment]
Marking this one as a draft as the `TestServer_Query_ExactTimeRange` InfluxQL test started newly failing after this change.

File: data_types/src/timestamp.rs [hunk 3]
```diff
@@ -27,11 +34,34 @@ impl TimestampRange {
...
+    }
 
     #[inline]
-    /// Returns if this range is disjoint w.r.t the provided range
-    pub fn disjoint(&self, other: &Self) -> bool {
-        self.end <= other.start || self.start >= other.end
+    /// Returns true if any of the values between min / max
+    /// (inclusive) are within the specified timestamp range if
+    /// self.contains(v_min) || self.contains(v_max)
+    pub fn contains(&self, range: TimestampRange) -> bool {
```
Comment (e-dard) [inline data_types/src/timestamp.rs:61]: I don't have strong opinions on this but in other projects we have tended to call this "overlaps".

File: data_types/src/timestamp.rs [hunk 4]
```diff
@@ -64,16 +94,33 @@ mod tests {
...
-
-        assert!(r1.disjoint(&r2));
-        assert!(r2.disjoint(&r1));
-        assert!(!r1.disjoint(&r3));
-        assert!(!r3.disjoint(&r1));
-        assert!(!r2.disjoint(&r3));
-        assert!(!r3.disjoint(&r2));
+    fn test_timestamp_range_contains_min_max() {
+        let range = TimestampRange::new(100, 200);
+        assert!(!TimestampMinMax::new(0, 99).contains(range));
```
Comment (e-dard) [inline data_types/src/timestamp.rs:99]: These look good 👍

**Review by e-dard — APPROVED** [review]
LGTM 👍

File: data_types/src/timestamp.rs (hunk 3 above)
Comment (alamb) [reply data_types/src/timestamp.rs]: renamed to `overlaps` in b4b7855a

**Pull request merged**